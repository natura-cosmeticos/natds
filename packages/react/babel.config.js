const path = require('path')

const defaultAlias = {
  '@naturacosmeticos/natds-react': path.resolve(__dirname, './src/components')
}

module.exports = {
  plugins: [
    ['babel-plugin-module-resolver', { alias: defaultAlias }]
  ],
  presets: [
    ['@babel/preset-env', { targets: { node: 'current' } }],
    '@babel/preset-typescript',
    '@babel/preset-react'
  ]
}
