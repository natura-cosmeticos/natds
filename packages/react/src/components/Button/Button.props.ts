import { IconName } from '@naturacosmeticos/natds-icons'

interface ButtonBaseProps {
  /**
   * If `true`, the button will be disabled.
   * @default false
   */
  disabled?: boolean;

  /**
   * If `true`, the button will occupy 100% of the width
   * @default false
   */
  fullWidth?: boolean;

  /**
   * Click handler
   */
  onClick: () => void;

  /**
  * The size of the button
  * @default semiX
  */
  size?: 'semi' | 'semiX' | 'medium';

  /**
   * Optional ID for testing
   */
  testID?: string;

  /**
   * The text content of the button
   * @default button
   */
  text: string;

  /**
   * The variant to use.
   * @default contained
   */
  variant?: 'contained' | 'outlined' | 'text';

}

type ButtonWithIcon = ButtonBaseProps & {
  /**
   * If `true`, will show an icon on the right
   * @default false
   */
  showIcon: true

  /**
   * Set the icon position
   */
  iconPosition: 'left' | 'right'

  /**
   * Set the icon to be rendered.
   * Check all available names in [Icon Library](https://ds.natura.design/28db352be/p/94367e-icon-library/b/6154b9)
   * @default outlined-default-mockup
   */
  iconName: IconName
}

export type ButtonProps = ButtonBaseProps & { showIcon?: never } | ButtonWithIcon
