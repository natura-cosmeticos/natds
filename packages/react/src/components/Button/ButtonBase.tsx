import React from 'react'
import { Size, Theme } from '@naturacosmeticos/natds-themes'
import { createUseStyles } from 'react-jss'
import { Ripple } from '../Ripple'

export interface ButtonBaseProps {
  classes?: string
  disabled?: boolean
  fullWidth?: boolean
  onClick: (e?: any) => void
  size: keyof Size
  testID?: string
  value?: string
  children: React.ReactNode
}

const styles = createUseStyles((theme: Theme) => ({
  base: {
    border: 'none',
    cursor: 'pointer',
    height: ({ size }: ButtonBaseProps) => size && theme.size[size],
    outline: 0,
    width: ({ fullWidth }: ButtonBaseProps) => (fullWidth ? '100%' : 'auto'),
    '&:disabled': {
      cursor: 'default'
    }
  }
}))

const ButtonBase = ({
  children,
  classes,
  disabled,
  fullWidth = false,
  onClick,
  size,
  testID,
  value
}: ButtonBaseProps): JSX.Element => {
  const { base } = styles({ fullWidth, size })

  return (
    <Ripple disabled={disabled} fullWidth={fullWidth}>
      <button
        className={`${base} ${classes}`}
        data-testid={testID}
        disabled={disabled}
        onClick={onClick}
        type="button"
        value={value}
      >
        {children}
      </button>
    </Ripple>
  )
}

export default ButtonBase
