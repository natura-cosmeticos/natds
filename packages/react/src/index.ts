export * from './ThemeProvider'
export { useTheme, createUseStyles } from 'react-jss'

// Components
export * from './components/Avatar'
export * from './components/Badge'
export * from './components/Button'
export * from './components/Card'
export * from './components/Checkbox'
export * from './components/Counter'
export * from './components/Dialog'
export * from './components/Divider'
export * from './components/Icon'
export * from './components/IconButton'
export * from './components/Input'
export * from './components/Label'
export * from './components/Link'
export * from './components/ListItem'
export * from './components/Logo'
export * from './components/ProgressIndicator'
export * from './components/RadioButton'
export * from './components/Rating'
export * from './components/TextField'
