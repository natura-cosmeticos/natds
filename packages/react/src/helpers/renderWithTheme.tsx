import React from 'react'
import { JssProvider, SheetsRegistry } from 'react-jss'
import { render } from '@testing-library/react'
import { ThemeProvider } from '../ThemeProvider'

const renderWithTheme = (component) => {
  const sheets = new SheetsRegistry()

  const renderTree = render(
    <JssProvider registry={sheets}>
      <ThemeProvider>
        {component}
      </ThemeProvider>
    </JssProvider>
  )

  return { styles: sheets.toString(), component: renderTree }
}

export default renderWithTheme
