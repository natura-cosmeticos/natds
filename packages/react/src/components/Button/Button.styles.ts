import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { ButtonProps } from './Button.props'

type ButtonStyleProps = Required<Pick<ButtonProps, 'size' | 'variant' | 'fullWidth' | 'disabled'>>

const isContained = ({ variant }: ButtonStyleProps) => variant === 'contained'
const isOutlined = ({ variant }: ButtonStyleProps) => variant === 'outlined'
const isFullWidth = ({ fullWidth }: ButtonStyleProps) => fullWidth
const isDisabled = ({ disabled }: ButtonStyleProps) => disabled

const getLabelStyles = (theme: Theme, props: ButtonStyleProps) => {
  const color = {
    active: (isContained(props) ? theme.color.onPrimary : theme.color.highEmphasis),
    disabled: ((isContained(props) && isDisabled(props))
      ? theme.color.highEmphasis : theme.color.mediumEmphasis
    )
  }
  return isDisabled(props) ? color.disabled : color.active
}

const styles = createUseStyles((theme: Theme) => ({
  button: {
    overflow: 'hidden',
    position: 'relative',
    backgroundColor: (props) => (isContained(props) ? theme.color.primary : 'transparent'),
    border: (props) => (isOutlined(props) ? `1px solid ${theme.color.primary}` : 0),
    borderRadius: 4,
    cursor: 'pointer',
    height: ({ size }: ButtonStyleProps) => theme.size[size],
    outline: 0,
    padding: [theme.spacing.tiny, theme.spacing.small],
    width: (props) => (isFullWidth(props) ? '100%' : 'auto'),
    '&:disabled': {
      cursor: 'default',
      backgroundColor: (props) => (isContained(props) ? theme.color.lowEmphasis : 'transparent'),
      border: (props) => (isOutlined(props) && `1px solid ${theme.color.lowEmphasis}`)
    }
  },
  overlay: {
    position: 'absolute',
    backgroundColor: theme.color.highlight,
    height: '100%',
    visibility: 'visible',
    width: '100%',
    top: 0,
    left: 0,
    opacity: 0,
    '&:hover': {
      opacity: theme.opacity.lower
    }
  },
  label: {
    fontWeight: 500,
    letterSpacing: 1.23,
    fontFamily: [theme.typography.fontFamily.primary, theme.typography.fontFamily.secondary],
    textTransform: 'uppercase',
    color: (props) => getLabelStyles(theme, props)
  }
}))

export default styles
