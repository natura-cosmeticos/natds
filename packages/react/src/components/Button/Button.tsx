import React from 'react'
import { ButtonProps } from './Button.props'
import { Ripple } from '../Ripple'
import styles from './Button.styles'

const Button = ({
  disabled = false,
  fullWidth = false,
  label,
  onClick,
  size = 'semiX',
  testID,
  variant = 'contained'
}: ButtonProps): JSX.Element => {
  const classes = styles({
    size, fullWidth, variant, disabled
  })

  return (
    <Ripple disabled={disabled} fullWidth={fullWidth}>
      <button
        className={classes.button}
        data-testid={testID}
        disabled={disabled}
        onClick={onClick}
        type="button"
      >
        <span className={classes.label}>{label}</span>
      </button>
    </Ripple>
  )
}

export default Button
