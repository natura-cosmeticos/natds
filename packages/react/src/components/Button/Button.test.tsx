import * as React from 'react'
import { fireEvent } from '@testing-library/react'
import Button from '.'
import renderWithTheme from '../../helpers/renderWithTheme'

jest.mock('../Ripple/Ripple')

describe('Button component', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<Button label="button" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should call onClick', () => {
    const onClickMock = jest.fn()
    const { component: { getByTestId } } = renderWithTheme(<Button label="button" onClick={onClickMock} testID="btn-test" />)

    fireEvent.click(getByTestId('btn-test'))

    expect(onClickMock).toHaveBeenCalled()
  })

  it('should render correctly when the size is semi', () => {
    const { styles, component } = renderWithTheme(<Button label="button" size="semi" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when the size is medium', () => {
    const { styles, component } = renderWithTheme(<Button label="button" size="medium" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when the fullWidth is true', () => {
    const { styles, component } = renderWithTheme(<Button label="button" fullWidth />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly when the variant is contained and disabled is true', () => {
    const { styles, component } = renderWithTheme(<Button label="button" disabled />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when the variant is outlined', () => {
    const { styles, component } = renderWithTheme(<Button label="button" variant="outlined" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when the variant is text', () => {
    const { styles, component } = renderWithTheme(<Button label="button" variant="text" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly when the variant is text and is disabled', () => {
    const { styles, component } = renderWithTheme(<Button label="button" variant="text" disabled />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when the variant is outlined and is disabled', () => {
    const { styles, component } = renderWithTheme(<Button label="button" variant="outlined" disabled />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
