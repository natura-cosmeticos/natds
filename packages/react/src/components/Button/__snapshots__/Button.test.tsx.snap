// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly when the fullWidth is true 1`] = `
Array [
  ".button-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  border-radius: 4px;
}
.button-0-2-1:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0;
  position: absolute;
  background-color: #000000;
}
.button-0-2-1:hover:not([disabled]):after {
  opacity: 0.04;
}
.button-0-2-1:focus:not([disabled]):after {
  opacity: 0.12;
}
.button-0-2-1:disabled {
  cursor: default;
  box-shadow: none;
}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  overflow: hidden;
  font-size: 14px;
  text-align: center;
  font-family: Roboto, sans-serif;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  letter-spacing: 1.23px;
  text-transform: uppercase;
}
.button-d0-0-2-4 {
  background-color: #f4ab34;
  border: 0;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14);
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: 100%;
}
.button-d0-0-2-4:disabled {
  background-color: #bbbbbb;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row;
}
.label-d2-0-2-6 {
  color: #333333;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="button-0-2-1 button-d0-0-2-4"
        type="button"
      >
        <span
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the size is medium 1`] = `
Array [
  ".button-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  border-radius: 4px;
}
.button-0-2-1:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0;
  position: absolute;
  background-color: #000000;
}
.button-0-2-1:hover:not([disabled]):after {
  opacity: 0.04;
}
.button-0-2-1:focus:not([disabled]):after {
  opacity: 0.12;
}
.button-0-2-1:disabled {
  cursor: default;
  box-shadow: none;
}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  overflow: hidden;
  font-size: 14px;
  text-align: center;
  font-family: Roboto, sans-serif;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  letter-spacing: 1.23px;
  text-transform: uppercase;
}
.button-d0-0-2-4 {
  background-color: #f4ab34;
  border: 0;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14);
  height: 48px;
  padding-left: 16px;
  padding-right: 16px;
  width: auto;
}
.button-d0-0-2-4:disabled {
  background-color: #bbbbbb;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row;
}
.label-d2-0-2-6 {
  color: #333333;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="button-0-2-1 button-d0-0-2-4"
        type="button"
      >
        <span
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the size is semi 1`] = `
Array [
  ".button-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  border-radius: 4px;
}
.button-0-2-1:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0;
  position: absolute;
  background-color: #000000;
}
.button-0-2-1:hover:not([disabled]):after {
  opacity: 0.04;
}
.button-0-2-1:focus:not([disabled]):after {
  opacity: 0.12;
}
.button-0-2-1:disabled {
  cursor: default;
  box-shadow: none;
}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  overflow: hidden;
  font-size: 14px;
  text-align: center;
  font-family: Roboto, sans-serif;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  letter-spacing: 1.23px;
  text-transform: uppercase;
}
.button-d0-0-2-4 {
  background-color: #f4ab34;
  border: 0;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14);
  height: 32px;
  padding-left: 4px;
  padding-right: 4px;
  width: auto;
}
.button-d0-0-2-4:disabled {
  background-color: #bbbbbb;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row;
}
.label-d2-0-2-6 {
  color: #333333;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="button-0-2-1 button-d0-0-2-4"
        type="button"
      >
        <span
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the variant is contained and disabled is true 1`] = `
Array [
  ".button-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  border-radius: 4px;
}
.button-0-2-1:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0;
  position: absolute;
  background-color: #000000;
}
.button-0-2-1:hover:not([disabled]):after {
  opacity: 0.04;
}
.button-0-2-1:focus:not([disabled]):after {
  opacity: 0.12;
}
.button-0-2-1:disabled {
  cursor: default;
  box-shadow: none;
}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  overflow: hidden;
  font-size: 14px;
  text-align: center;
  font-family: Roboto, sans-serif;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  letter-spacing: 1.23px;
  text-transform: uppercase;
}
.button-d0-0-2-4 {
  background-color: #f4ab34;
  border: 0;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14);
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.button-d0-0-2-4:disabled {
  background-color: #bbbbbb;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row;
}
.label-d2-0-2-6 {
  color: #333333;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="button-0-2-1 button-d0-0-2-4"
        disabled=""
        type="button"
      >
        <span
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the variant is outlined 1`] = `
Array [
  ".button-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  border-radius: 4px;
}
.button-0-2-1:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0;
  position: absolute;
  background-color: #000000;
}
.button-0-2-1:hover:not([disabled]):after {
  opacity: 0.04;
}
.button-0-2-1:focus:not([disabled]):after {
  opacity: 0.12;
}
.button-0-2-1:disabled {
  cursor: default;
  box-shadow: none;
}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  overflow: hidden;
  font-size: 14px;
  text-align: center;
  font-family: Roboto, sans-serif;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  letter-spacing: 1.23px;
  text-transform: uppercase;
}
.button-d0-0-2-4 {
  background-color: transparent;
  border: 1px solid #f4ab34;
  box-shadow: none;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.button-d0-0-2-4:disabled {
  background-color: transparent;
  border: 1px solid #bbbbbb;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row;
}
.label-d2-0-2-6 {
  color: #333333;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="button-0-2-1 button-d0-0-2-4"
        type="button"
      >
        <span
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the variant is outlined and is disabled 1`] = `
Array [
  ".button-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  border-radius: 4px;
}
.button-0-2-1:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0;
  position: absolute;
  background-color: #000000;
}
.button-0-2-1:hover:not([disabled]):after {
  opacity: 0.04;
}
.button-0-2-1:focus:not([disabled]):after {
  opacity: 0.12;
}
.button-0-2-1:disabled {
  cursor: default;
  box-shadow: none;
}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  overflow: hidden;
  font-size: 14px;
  text-align: center;
  font-family: Roboto, sans-serif;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  letter-spacing: 1.23px;
  text-transform: uppercase;
}
.button-d0-0-2-4 {
  background-color: transparent;
  border: 1px solid #f4ab34;
  box-shadow: none;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.button-d0-0-2-4:disabled {
  background-color: transparent;
  border: 1px solid #bbbbbb;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row;
}
.label-d2-0-2-6 {
  color: #777777;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="button-0-2-1 button-d0-0-2-4"
        disabled=""
        type="button"
      >
        <span
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the variant is text 1`] = `
Array [
  ".button-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  border-radius: 4px;
}
.button-0-2-1:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0;
  position: absolute;
  background-color: #000000;
}
.button-0-2-1:hover:not([disabled]):after {
  opacity: 0.04;
}
.button-0-2-1:focus:not([disabled]):after {
  opacity: 0.12;
}
.button-0-2-1:disabled {
  cursor: default;
  box-shadow: none;
}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  overflow: hidden;
  font-size: 14px;
  text-align: center;
  font-family: Roboto, sans-serif;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  letter-spacing: 1.23px;
  text-transform: uppercase;
}
.button-d0-0-2-4 {
  background-color: transparent;
  border: 0;
  box-shadow: none;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.button-d0-0-2-4:disabled {
  background-color: transparent;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row;
}
.label-d2-0-2-6 {
  color: #333333;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="button-0-2-1 button-d0-0-2-4"
        type="button"
      >
        <span
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the variant is text and is disabled 1`] = `
Array [
  ".button-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  border-radius: 4px;
}
.button-0-2-1:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0;
  position: absolute;
  background-color: #000000;
}
.button-0-2-1:hover:not([disabled]):after {
  opacity: 0.04;
}
.button-0-2-1:focus:not([disabled]):after {
  opacity: 0.12;
}
.button-0-2-1:disabled {
  cursor: default;
  box-shadow: none;
}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  overflow: hidden;
  font-size: 14px;
  text-align: center;
  font-family: Roboto, sans-serif;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  letter-spacing: 1.23px;
  text-transform: uppercase;
}
.button-d0-0-2-4 {
  background-color: transparent;
  border: 0;
  box-shadow: none;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.button-d0-0-2-4:disabled {
  background-color: transparent;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row;
}
.label-d2-0-2-6 {
  color: #777777;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="button-0-2-1 button-d0-0-2-4"
        disabled=""
        type="button"
      >
        <span
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly with default props 1`] = `
Array [
  ".button-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  border-radius: 4px;
}
.button-0-2-1:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0;
  position: absolute;
  background-color: #000000;
}
.button-0-2-1:hover:not([disabled]):after {
  opacity: 0.04;
}
.button-0-2-1:focus:not([disabled]):after {
  opacity: 0.12;
}
.button-0-2-1:disabled {
  cursor: default;
  box-shadow: none;
}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  overflow: hidden;
  font-size: 14px;
  text-align: center;
  font-family: Roboto, sans-serif;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  letter-spacing: 1.23px;
  text-transform: uppercase;
}
.button-d0-0-2-4 {
  background-color: #f4ab34;
  border: 0;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14);
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.button-d0-0-2-4:disabled {
  background-color: #bbbbbb;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row;
}
.label-d2-0-2-6 {
  color: #333333;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="button-0-2-1 button-d0-0-2-4"
        type="button"
      >
        <span
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly with the given iconName 1`] = `
Array [
  ".icon-0-2-7 {
  font-family: natds-icons;
  user-select: none;
}
.icon-d0-0-2-8 {
  color: #333333;
  font-size: 24px;
}
.button-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  border-radius: 4px;
}
.button-0-2-1:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0;
  position: absolute;
  background-color: #000000;
}
.button-0-2-1:hover:not([disabled]):after {
  opacity: 0.04;
}
.button-0-2-1:focus:not([disabled]):after {
  opacity: 0.12;
}
.button-0-2-1:disabled {
  cursor: default;
  box-shadow: none;
}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  overflow: hidden;
  font-size: 14px;
  text-align: center;
  font-family: Roboto, sans-serif;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  letter-spacing: 1.23px;
  text-transform: uppercase;
}
.button-d0-0-2-4 {
  background-color: #f4ab34;
  border: 0;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14);
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.button-d0-0-2-4:disabled {
  background-color: #bbbbbb;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row-reverse;
}
.label-d2-0-2-6 {
  color: #333333;
  margin-right: 8px;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="button-0-2-1 button-d0-0-2-4"
        type="button"
      >
        <span
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
          <i
            aria-hidden="true"
            class="icon-0-2-7 icon-d0-0-2-8 natds-icons natds-icons-filled-action-check"
            data-testid="icon-filled-action-check"
            role="img"
          />
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render icon to the left when the showIcon is true and iconPosition is left 1`] = `
Array [
  ".icon-0-2-7 {
  font-family: natds-icons;
  user-select: none;
}
.icon-d0-0-2-8 {
  color: #333333;
  font-size: 24px;
}
.button-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  border-radius: 4px;
}
.button-0-2-1:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0;
  position: absolute;
  background-color: #000000;
}
.button-0-2-1:hover:not([disabled]):after {
  opacity: 0.04;
}
.button-0-2-1:focus:not([disabled]):after {
  opacity: 0.12;
}
.button-0-2-1:disabled {
  cursor: default;
  box-shadow: none;
}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  overflow: hidden;
  font-size: 14px;
  text-align: center;
  font-family: Roboto, sans-serif;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  letter-spacing: 1.23px;
  text-transform: uppercase;
}
.button-d0-0-2-4 {
  background-color: #f4ab34;
  border: 0;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14);
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.button-d0-0-2-4:disabled {
  background-color: #bbbbbb;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row;
}
.label-d2-0-2-6 {
  color: #333333;
  margin-left: 8px;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="button-0-2-1 button-d0-0-2-4"
        type="button"
      >
        <span
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
          <i
            aria-hidden="true"
            class="icon-0-2-7 icon-d0-0-2-8 natds-icons natds-icons-outlined-default-mockup"
            data-testid="icon-outlined-default-mockup"
            role="img"
          />
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render icon to the right when the showIcon is true 1`] = `
Array [
  ".icon-0-2-7 {
  font-family: natds-icons;
  user-select: none;
}
.icon-d0-0-2-8 {
  color: #333333;
  font-size: 24px;
}
.button-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  border-radius: 4px;
}
.button-0-2-1:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0;
  position: absolute;
  background-color: #000000;
}
.button-0-2-1:hover:not([disabled]):after {
  opacity: 0.04;
}
.button-0-2-1:focus:not([disabled]):after {
  opacity: 0.12;
}
.button-0-2-1:disabled {
  cursor: default;
  box-shadow: none;
}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  overflow: hidden;
  font-size: 14px;
  text-align: center;
  font-family: Roboto, sans-serif;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  letter-spacing: 1.23px;
  text-transform: uppercase;
}
.button-d0-0-2-4 {
  background-color: #f4ab34;
  border: 0;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14);
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.button-d0-0-2-4:disabled {
  background-color: #bbbbbb;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row;
}
.label-d2-0-2-6 {
  color: #333333;
  margin-right: 8px;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="button-0-2-1 button-d0-0-2-4"
        type="button"
      >
        <span
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
          <i
            aria-hidden="true"
            class="icon-0-2-7 icon-d0-0-2-8 natds-icons natds-icons-outlined-default-mockup"
            data-testid="icon-outlined-default-mockup"
            role="img"
          />
        </span>
      </button>
    </div>
  </div>,
]
`;
